test/FunctionalTests/Tests/DataServices/UnitTests/DesignT4UnitTests/CodeGenerationContextTest.cs

```diff
@@ -294,5 +294,69 @@ public void GetEdmxStringFromMetadataPathShouldThrowExceptionIfInputIsInvalidUri
                 Assert.AreEqual(ex.Message, "Only file, http, https schemes are supported for paths to metadata source locations.");
             }
         }
+
+        [TestMethod]
+        public void SetOneCustomHeaderCorrectlyShouldNotThrowException()
+        {
+            var codegen = new ODataT4CodeGenerator();
+            string customHeaderString = @"Authorization:Bearer bearer-token";
+            codegen.ValidateAndSetCustomHttpHeadersFromString(customHeaderString);
+            Assert.IsNotNull(codegen.CustomHttpHeaders);
+            Assert.AreEqual(1, codegen.CustomHttpHeaders.Count);
+            Assert.AreEqual("Authorization:Bearer bearer-token", codegen.CustomHttpHeaders.FirstOrDefault());
+        }
+
+        [TestMethod]
+        public void SetMultipleCustomHeadersCorrectlyShouldNotThrowException()

```

Comment:
Would be nice to add a test for ```SetMultipleCustomHeadersWithEmptyNewLinesShouldEitherThrowOrNot()``` with a sample input like
```
var customHeaderString = @"
Auth: Yes

odata.continue-on-error: false
";
```
To document the choice of StringSplitOptions.None when splitting headerValue.

Reply:
Added